frontends/infinite-corridor/src/pages/SearchPage.tsx

```diff
@@ -1,11 +1,145 @@
+const SEARCH_RESUTLT_CELL_WIDTH = 12
+
+interface Result {
+  _source: LearningResourceResult
+}
+
+const search = async (params: SearchQueryParams) => {
+  const body = buildSearchQuery(params)
+
+  return fetchJSONWithCSRF("/api/v0/search/", {
+    method: "post",
+    body:   JSON.stringify(body)
+  })
+}
 
 const SearchPage: React.FC = () => {
-  const [searchParams] = useSearchParams()
+  const [results, setSearchResults] = useState<Result[]>([])
+  const [total, setTotal] = useState(0)
+  const [completedInitialLoad, setCompletedInitialLoad] = useState(false)
+  const [requestInFlight, setRequestInFlight] = useState(false)
+  const [searchApiFailed, setSearchApiFailed] = useState(false)
+  const [aggregations, setAggregations] = useState<Aggregations>(new Map())
+
+  const clearSearch = useCallback(() => {
+    setSearchResults([])
+    setCompletedInitialLoad(false)
+    setTotal(0)
+  }, [])
+
+  const runSearch = useCallback(
+    async (text: string, activeFacets: Facets, from: number) => {
+      if (activeFacets["type"]) {
+        activeFacets["type"] = intersection(activeFacets["type"], ALLOWED_TYPES)
+      } else {
+        activeFacets["type"] = ALLOWED_TYPES
+      }
+      setRequestInFlight(true)
+      const newResults = await search({
+        text,
+        from,
+        activeFacets,
+        size: pageSize
+      })
+      setRequestInFlight(false)
+
+      if (!newResults || newResults["apiFailed"]) {
+        setSearchApiFailed(true)
+        return
+      }
+
+      setAggregations(new Map(Object.entries(newResults.aggregations ?? {})))
+
+      setSearchResults(results =>
+        from === 0 ?
+          newResults.hits.hits :
+          [...results, ...newResults.hits.hits]
+      )
+      setTotal(newResults.hits.total)
+      setCompletedInitialLoad(true)
+    },
+    []
+  )
+
+  const { updateText, loadMore, text, onSubmit, from, clearText } =
+    useCourseSearch(
+      runSearch,
+      clearSearch,
+      aggregations,
+      // this is the 'loaded' value, which is what useCourseSearch uses
+      // to determine whether to fire off a request or not.
+      completedInitialLoad && !requestInFlight,
+      pageSize
+    )
+
   return (
-    <div className="page-content">
-      Search params: <code>{searchParams.toString()}</code>
+    <div>
+      <BannerPageWrapper>
+        <BannerPageHeader tall compactOnMobile>
+          <Searchbox
+            className="course-searchbox"
+            onChange={updateText}
+            value={text || ""}
+            onClear={clearText}
+            //@ts-expect-error - types need to be fixed in course-search-utils
+            onSubmit={onSubmit}
+            autoFocus
+          />
+        </BannerPageHeader>
+      </BannerPageWrapper>
+      <InfiniteScroll
+        hasMore={from + pageSize < total}
+        loadMore={loadMore}
+        initialLoad={from === 0}
+        loader={
+          <div className="loader" key={0}>
```

Comment:
```suggestion
          <div className="loader">
```

Reply:
This is needed for the infinite scroll. I get
```
Warning: Each child in a list should have a unique "key" prop.

Check the render method of `InfiniteScroll`. It was passed a child from SearchPage. See https://fb.me/react-warning-keys for more information.
    in div (created by SearchPage)
    in InfiniteScroll (created by SearchPage)
    in div (created by SearchPage)
    in SearchPage (created by App)
    in Route (created by App)
    in Router (created by App)
    in QueryClientProvider (created by App)
    in Fe (created by App)
    in div (created by App)
    in App
```
If I remove it